### ISSUE TITLE
Validation ppl lower than train ppl

### ISSUE THREAD

**Issue by vrublack** [quanpn90/NMTGMinor#1]
I'm using the transformer from commit 6b9a081 and my validation perplexity is lower than my train perplexity. I don't think this is a dataset issue. Is there maybe a bug? Thanks!

**Comment by quanpn90** [comment]
If validation perplexity is lower than train perplexity, the model is doing well isn't it ?

Because you're not overfitting.

**Comment by vrublack** [comment]
It's not overfitting but it's still strange. Although dropout is deactivated in evaluation so maybe that's an explanation?

**Comment by quanpn90** [comment]
Can you explain how strange it is ?

Normally when we train NNs, we don't want the train perplexity to be < valid perplexity.  If your train perplexity is too high, the model might be struggling to learn

**Comment by vrublack** [comment]
Usually, if training goes well the valid ppl is slightly higher than train ppl but not much higher. Because it has already seen the train set but not the valid set, therefore it should be slightly easier for it to predict the train set, even if it's generalizing well.

**Comment by vrublack** [comment]
Here is a related discussion: https://stats.stackexchange.com/questions/187335/validation-error-less-than-training-error

**Comment by quanpn90** [comment]
I don't know what your problem / data is, but there shouldn't be any relation between training and validation perplexity, in general.

As one already pointed out, there could be the case of:

Your training set had many 'hard' cases to learn
Your validation set had mostly 'easy' cases to predict

The valid set is not seen, but are you sure that it's totally different than your training set ? 
What you really want to measure during training, is the relative rate of perplexity reduction between training and validation over time. If training perplexity keeps decreasing, and valdation ppl also decreases, your model is learning well and can still generalize (to some extent, because overfitting is just relatively assessed). Otherwise we need to stop training when there's no progress on validation.

**Comment by vrublack** [comment]
Thanks, I'll see what happens if I turn off dropout

**Comment by vrublack** [comment]
Turned off dropout -> train and valid loss are now the same, so dropout seemed to have been the cause
